packages/@okta/vuepress-site/docs/guides/terraform-design-access-security/main/index.md

```diff
@@ -0,0 +1,194 @@
+layout: Guides
+---
+
+Terraform requires permissions to read and manage objects in your org. You can configure these permissions in multiple ways, depending on what you want Terraform to automate.
+
+Be deliberate with Terraform permissions to keep your org secure. Terraform doesn’t need access to your entire org. Instead, limit Terraform to access only the Okta objects that you want Terraform to manage. This is known as *least-privilege* access, which reduces the risk of unintended changes and misuse.
+
+## Select an authorization method to control access
+
+Okta offers multiple ways to authorize Terraform. Okta recommends using the most secure method that works for your use case.
+
+### OAuth 2.0
+
+The OAuth 2.0 protocol provides a high level of security for authorizing Terraform. With OAuth 2.0, you can customize granular access to Okta resources, limit the time of access, and revoke access. These features give you control over your org’s security.
+
+See [OAuth 2.0 and OpenID Connect Overview](/docs/concepts/oauth-openid/).
+
+### API tokens
+
+API tokens are easy to implement but can reduce your org security. The permissions of an API token are the same as those of the admin that creates it. The permissions for an API token can't be changed after it's created. Also, you must manually delete a token. This combination leaves API tokens vulnerable to misuse and exploitation.
+
+If you use API tokens, create the token with a service account that has only the permissions required for your Terraform configuration.
+
+This guide focuses only on OAuth 2.0 authorization, because of the security benefits.
+
+## Authorize Terraform with OAuth 2.0
+
+Configure OAuth 2.0 access by creating an Okta app that's used by Terraform.
+
+> **Note:** Not every resource in the Okta Provider supports OAuth 2.0. See the [Okta Terraform Provider documentation](https://registry.terraform.io/providers/okta/okta/latest/docs).
+
+To implement OAuth 2.0 authorization for Terraform, see [Enable Terraform access for your Okta org](/docs/guides/terraform-enable-org-access).
+
+### Configure an app for authorization
+
+Use an Okta API service app for machine-to-machine communication with Terraform. A service app authorizes Terraform using the [Client Credentials flow](/docs/guides/implement-grant-type/clientcreds/main/).
+
+With this flow, Terraform uses credentials to request access to your org. The service app verifies the credentials and authorizes Terraform access to Okta.
+
+The credentials used for this flow are a public/private key pair. Okta stores the public key in the service app, and Terraform uses the private key in the configuration. You can [generate the key pair] in the Okta service app or with your own secure internal methods. See [Create access credentials](/docs/guides/terraform-enable-org-access#create-access-credentials) in [Enable Terraform access for your Okta org](/docs/guides/terraform-enable-org-access).
```

Comment:
Looks like `[generate the key pair]` is missing the url

Reply:
Thanks... I put in a link to an article on implementing OAuth for a service app... I'm not 100% sure of the link as I didn't write this.